networking/routetable.go

```diff
@@ -176,44 +176,55 @@ func (table *routingTable) findPreferredRoute() (ok bool, ifname string, route *
 	return
 }
 
-func (table *routingTable) setPreferredRoute(removeDefaultRoute bool) (err error) {
+func (table *routingTable) setPreferredRoute(overrideDefaultRoute bool) (err error) {
 	table.lock.Lock()
 	// was their an old route, remove it.
-	if table.oldDefaultRoute != nil { //} && (table.oldDefaultRoute != table.defaultRoute) {
+	if table.oldDefaultRoute != nil && overrideDefaultRoute { //} && (table.oldDefaultRoute != table.defaultRoute) {
```

Comment:
When is oldDefaultRoute set and what is it set to?

Reply:
oldDefaultRoute contains the default route that was previously set by maestro. (i.e the default route that was set last time network configuration was changed in maestro)

Reply:
Where is table.oldDefaultRoute is assigned?  I was looking for it to be assigned in this function, but it's not assigned here so I'm guessing it's somewhere else.  I'm trying to understand the circumstances and context of this function deleting the route (line 184).

Reply:
https://github.com/armPelionEdge/maestro/blob/1438eb113e27e044f9ba03d0d461ece3b82d32ae/networking/routetable.go#L162

Reply:
I think I may have made a mistake. Should the oldDefaultRoute be deleted no matter what the flag is set to?

Reply:
Probably want to leave it unchanged unless we're replacing it with something else.

Reply:
> https://github.com/armPelionEdge/maestro/blob/1438eb113e27e044f9ba03d0d461ece3b82d32ae/networking/routetable.go#L162

Help me understand the context.  What is it about this old route that maestro decided to mark it for deletion?  Why does that determination occur in a Find-like function?  What is the relationship of the Find-like function with this Set-like function.

Reply:
@costanic Can we have a quick call to discuss this?